alacritty/src/input/keyboard.rs

```diff
@@ -644,3 +643,11 @@ impl From<ModifiersState> for SequenceModifiers {
         modifiers
     }
 }
+
+/// Check whether the text represents a C0 or C1 codepoint.
+fn is_c0c1_codepoint(text: &str) -> bool {
+    let codepoint = text.bytes().next().unwrap();
+    // 0x7f (DEL) is included here since it has a dedicated control code (`^?`) which generally
+    // does not match the reported text (`^H`), despite not technically being part of C0 or C1.
+    text.bytes().len() == 1 && (codepoint < 0x20 || (0x7f..=0x9f).contains(&codepoint))
```

Comment:
Is there any difference between `text.bytes().len()` and `text.len()`? Isn't `text.len()` the byte length anyway?

And even if it was the character length instead, it would still work since the only characters that start with the control codes are the C0 and C1s.

Reply:
No, there's no.

The issue is that C0/C1 could be a part of the large compose insert, no? But I guess we should throw it away anyway.

Reply:
I have no problem with checking for length here. I think it makes sense and I don't think we should report this as a control character when the length is bigger than 1.

All I was saying is that you can change `text.bytes().len()` to `text.len()`.